iceoryx_utils/include/iceoryx_utils/cxx/deadlinetimer.hpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_UTILS_CXX_DEADLINETIMER_HPP
+#define IOX_UTILS_CXX_DEADLINETIMER_HPP
+
+#include "iceoryx_utils/internal/units/duration.hpp"
+
+#include <chrono>
+#include <cstdint>
+
+namespace iox
+{
+namespace cxx
+{
+/// @brief This offers the deadline timer functionality. It has user convenient methods to reset the timer [by default
+/// it uses the intialized duration], reset timer to a customized duration, check if the timer is active and user can
+/// also get to know about the remaining time before the timer goes off
+/// @code
+///     iox::clock::DeadlineTimer deadlineTimer(1000_ms);
```

Comment:
Should be `iox::cxx::DeadlineTimer` now.

Reply:
yes this is modified